std/math.d

```diff
@@ -5884,14 +5885,285 @@ T nextafter(T)(const T x, const T y) @safe pure nothrow @nogc
 real fdim(real x, real y) @safe pure nothrow @nogc { return (x > y) ? x - y : +0.0; }
 
 /****************************************
- * Returns the larger of x and y.
+ * Returns the largest (most positive) element of `nums`.
+ *
+ * Elements are compared after implicit conversion to a common floating-point
+ * type. If `nums` is empty or any element `isNaN`, a `nan` value will be returned.
  */
-real fmax(real x, real y) @safe pure nothrow @nogc { return x > y ? x : y; }
+auto fmax(T ...)(T nums)
+    if (allSatisfy!(ApplyRight!(isImplicitlyConvertible, real), T))
+{
+    pragma(inline, true);
+
+    alias F = AutoFImumType!T;
+    static if (T.length <= 4)
+        return fimumImpl!(">", T.length, F)(nums);
+    else
+        return fimumImpl!(">", F)(nums);
+}
+
+/// ditto
+auto fmax(F, T ...)(T nums)
+    if (isFloatingPoint!F && allSatisfy!(ApplyRight!(isImplicitlyConvertible, F), T))
+{
+    pragma(inline, true);
+
+    alias R = OriginalType!(Unqual!F);
+    static if (T.length <= 4)
+        return fimumImpl!(">", T.length, R)(nums);
+    else
+        return fimumImpl!(">", R)(nums);
+}
+
+/// ditto
+auto fmax(F)(const F[] nums) pure @safe nothrow @nogc
+    if (isFloatingPoint!F)
+{
+    pragma(inline, true);
+
+    alias R = OriginalType!(Unqual!F);
+    return fimumImpl!(">", R)(nums);
+}
+
+/// ditto
+auto fmax(F, size_t count)(auto ref const(F[count]) nums) pure @safe nothrow @nogc
+    if (isFloatingPoint!F)
+{
+    pragma(inline, true);
+
+    alias R = OriginalType!(Unqual!F);
+    static if (count <= 4)
+        return fimumImpl!(">", count, R)(nums);
+    else
+        return fimumImpl!(">", R)(nums[]);
+}
+
+///
+pure @safe nothrow @nogc unittest
+{
+    // Any number of arguments is supported:
+    assert(fmax(101.0, -5.0, 7.3723e121, double.epsilon) == 7.3723e121);
+    assert(fmax(1) == 1.0);
+    assert(fmax(ubyte.max, 25.0f) == 255.0f);
+    // Static and dynamic floating-point arrays, too:
+    double[7] nums = [0.0, double.min_normal, 0.5, -1.0, -double.max, double.infinity, 8];
+    assert(fmax(nums) == double.infinity);
+    assert(fmax(nums[1 .. 3]) == 0.5);
+
+    // NaN handling:
+    assert(fmax().isNaN);
+    assert(fmax(13, -431_315, long.max, real.nan, float.infinity).isNaN);
+
+    // The return type is always floating-point, and mixed signed-unsigned comparisons
+    // are safe, even if all the arguments are integers.
+    assert(fmax(int.min, cast(uint)int.max) is cast(double)int.max);
+    // The F template parameter can be used to request a specific floating-point type:
+    static assert(is(typeof(fmax!float(1u)) == float));
+    static assert(is(typeof(fmax!double(1L)) == double));
+    static assert(is(typeof(fmax!real(1.0f)) == real));
+}
```

Comment:
> std.complex

The concept of a maximum or minimum value is not well-defined for complex numbers. (That's why `std.complex.Complex` does not implement `opCmp()`.)

> std.bigint

That's outside the scope of `fmax()` and `fmin()`; as I state in the documentation, only types that can be implicitly converted to a floating-point type are supported.